neps/nep-0508.md

```diff
@@ -0,0 +1,231 @@
+
+### Removal of Fixed shards
+
+Fixed shards was a feature of the protocol that allowed for assigning specific accounts and all of their recursive sub accounts to a predetermined shard. This feature was only used for testing and was never used in production. Fixed shards fature unfortunately breaks the contiguity of shards and is not compatible with the new resharding flow. A sub account of a fixed shard account can fall in the middle of account range that belongs to a different shard. This property of fixed shards made it particularly hard to reason about and implement efficient resharding. 
+
+This was implemented ahead of this NEP and the fixed shards feature was **removed**. 
+
+### Garbage collection
+
+In epoch T+2 once resharding is completed, we can delete the trie state and the flat state related to the parent shard. In practice, this is handled as part of the garbage collection code. While garbage collecting the last block of epoch T+1, we go ahead and clear all the data associated with the parent shard from the trie cache, flat storage, and RocksDB state associated with trie state and flat storage.
+
+### Transaction pool
+
+The transaction pool is sharded i.e. it groups transactions by the shard where each transaction should be converted to a receipt. The transaction pool was previously sharded by the ShardId. Unfortunately ShardId is insufficient to correctly identify a shard across a resharding event as ShardIds change domain. The transaction pool was migrated to group transactions by ShardUId instead, and a transaction pool resharding was implemented to reassign transaction from parent shard to children shards right before the new shard layout takes effect. The ShardUId contains the version of the shard layout which allows differentiating between shards in different shard layouts. 
+
+This was implemented ahead of this NEP and the transaction pool is now fully **migrated** to ShardUId. 
+
+## Alternatives
+
+### Why is this design the best in the space of possible designs?
+
+This design is the simplest, most robust and safe while meeting all of the requirements. 
+
+### What other designs have been considered and what is the rationale for not choosing them?
```

Comment:
Perhaps this section should be titled, "What other implementation strategies have been considered ...", since these are not designs. At a high level perhaps the "other designs considered" would include things like, stopping the chain to do an offline resharding and resuming again; or why we decided that we would only split shards and not be able to merge them; why we decided to go with hardcoded resharding boundaries for now.